ddl/ddl_api.go

```diff
@@ -2373,7 +2373,7 @@ func (d *ddl) DropColumn(ctx sessionctx.Context, ti ast.Ident, spec *ast.AlterTa
 
 	// Check whether dropped column has existed.
 	colName := spec.OldColumnName.Name
-	col := table.FindCol(t.Cols(), colName.L)
+	col := table.FindCol(t.VisibleCols(), colName.L)
```

Comment:
We can't drop hide columns. Could we do `modify column` operation to hide columns?

Reply:
We can't modify the hidden column in theory, I will disable this operation in next PR